crates/node/src/lib.rs

```diff
@@ -504,41 +504,56 @@ impl Node {
                     error!(
                         %topic_hash,
                         %their_peer_id,
-                        "observed subscription to non-context topic, ignoring.."
-                    );
-
-                    return Ok(());
-                };
-
-                let Some(context) = self.ctx_manager.get_context(&context_id)? else {
-                    error!(
-                        %context_id,
-                        %their_peer_id,
-                        "observed subscription to unknown context, ignoring.."
+                        "Failed to parse topic hash into context ID, ignoring.."
                     );
 
                     return Ok(());
                 };
 
                 if self
                     .ctx_manager
-                    .is_application_installed(&context.application_id)
+                    .is_context_pending_catchup(&context_id)
```

Comment:
why not leave it to the new subscriber to make an explicit request to catchup?

and make the catchup process bidirectional?

so:

node A joins the context when nobody is online
node B comes online, subscribes - notifying node A
nothing happens just yet
node B sends a catchup request to all who may be available
node A responds, but since he lags behind node B
A stream is opened for node B to aid node A catchup

Reply:
And the reason for this, is.. what if node B helps node A get caught up, but node B is itself behind

when node C which has the latest state comes online, neither node A or B will know to make a catchup request

Reply:
> there are no mesh peers,

good point..

> make a direct request (WhatIsYourLatestTransactionHash)

that can be an accommodation for this, unless there's some other mechanism (event) that indicates mesh peer connection

Reply:
> > that can be an accommodation for this, unless there's some other mechanism (event) that indicates mesh peer connection
>
> Subscribed event indicates new mesh peer connection (and we are already reacting to it)

when node C subscribes, it doesn't notify node C about node A and node B's prior subscription

so node C won't know to initiate the catchup request

I was hoping for another mechanism entirely, although that might require manual implementation (since it's eventually possible to know how many other peers you're connected to are subscribed to a topic)

> how about we try to perform a catchup whenever we get a Subscribed event?

my concern is the potential for all other peers to immediately spam you with catchup requests which I'd expect to be concerning

as opposed to the requesting node iteratively requesting a catchup payload in an iterative process